client/my-sites/theme/theme-style-variations/index.tsx

```diff
@@ -0,0 +1,99 @@
+import { Button } from '@automattic/components';
+import { PremiumBadge } from '@automattic/design-picker';
+import { useLayoutEffect, useRef, useState } from '@wordpress/element';
+import classNames from 'classnames';
+import { translate } from 'i18n-calypso';
+import AsyncLoad from 'calypso/components/async-load';
+import type { StyleVariation } from '@automattic/design-picker/src/types';
+import type { TranslateResult } from 'i18n-calypso';
+
+import './style.scss';
+
+interface ThemeStyleVariationsProps {
+	description: TranslateResult;
+	selectedVariation: StyleVariation;
+	variations: StyleVariation[];
+	onClick: ( variation: StyleVariation ) => void;
+}
+
+const ThemeStyleVariations = ( {
+	description,
+	selectedVariation,
+	variations,
+	onClick,
+}: ThemeStyleVariationsProps ) => {
+	const observerRef = useRef< HTMLDivElement | null >( null );
+	const [ collapsibleMaxHeight, setCollapsibleMaxHeight ] = useState< number | undefined >();
+	const [ isCollapsible, setIsCollapsible ] = useState< boolean >( false );
+	const [ isCollapsed, setIsCollapsed ] = useState< boolean >( false );
+
+	useLayoutEffect( () => {
+		if ( ! observerRef.current ) {
+			return;
+		}
+
+		const resizeObserver = new ResizeObserver( ( [ observerNode ] ) => {
+			const node = observerNode.target;
+			const nodeFirstChildRect = ( node.firstChild as HTMLElement )?.getBoundingClientRect();
+			const nodeLastChildRect = ( node.lastChild as HTMLElement )?.getBoundingClientRect();
+			if ( ! nodeFirstChildRect || ! nodeLastChildRect ) {
+				return;
+			}
+
+			// Detect flex wrap.
+			const currentIsCollapsible = nodeFirstChildRect.top !== nodeLastChildRect.top;
```

Comment:
[nit] we can check the offsetTop of the last element to detect whether flex wrap and then we don't need to call `getBoundingClientRect` all the time.

```js
const currentIsCollapsible = node.lastChild.offsetTop > paddingTop
```

Reply:
I'm not sure what you meant with `paddingTop` so I just changed it to compare the `offsetTop` of `firstChild` and `lastChild`. We still manage to avoid using `getBoundingClientRect()` this way!

Reply:
Oh yes, that also makes senses 👍